components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -1166,21 +1166,44 @@ func (p *pg) GetStagedOrAppliedRule(ctx context.Context, projectID string, ruleI
 	return &rule, nil
 }
 
+func (p *pg) FetchAppliedRulesByProjectIDs(ctx context.Context) (map[string][]*v2.Rule, error) {
+	rules, err := p.listRulesUsingFunction(ctx, "SELECT query_rules($1)", false)
+	if err != nil {
+		return nil, err
+	}
+
+	// nothing stored while on v1 or v2.0
+	if len(rules) == 0 {
```

Comment:
I dunno if you need this optimization, since no work is being done below if `len(rules) == 0` and the object returned is the same. Could drop it I think.